### PR TITLE
#83 sync up with MicroProfile Config

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang@uk.ibm.com>

### CODE REVIEW

File: tck/src/main/java/org/eclipse/configjsr/CDIPlainInjectionTest.java [hunk 1]
```diff
@@ -65,13 +65,14 @@ public static Archive deployment() {
                 .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
     }
 
-    @AfterTest
-    public void tearDown() {
-        clear_all_property_values();
+    @BeforeTest
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/CDIPlainInjectionTest.java]: why BeforeTest? Seems wrong.

File: tck/src/main/java/org/eclipse/configjsr/CDIPlainInjectionTest.java [hunk 2]
```diff
@@ -88,37 +89,51 @@ public void can_inject_simple_values_when_defined() {
...
     @Test
-    public void can_inject_dynamic_values_via_CDI_provider() {
-        clear_all_property_values();
+    public void canInjectDynamicValuesViaCdiProvider() {
 
         DynamicValuesBean bean = getBeanOfType(DynamicValuesBean.class);
 
         //X TODO clarify how Provider<T> should behave for missing values assertThat(bean.getIntProperty(), is(nullValue()));
 
-        ensure_all_property_values_are_defined();
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/CDIPlainInjectionTest.java]: why? this should stay in there imo.

File: tck/src/main/java/org/eclipse/configjsr/CDIPlainInjectionTest.java [hunk 3]
```diff
@@ -189,21 +203,14 @@ private void clear_all_property_values() {
         @ConfigProperty(name="my.double.property")
         private double doubleProperty;
 
-        @Inject
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/CDIPlainInjectionTest.java]: yikes, seems you killed our new features which are not yet present in MicroProfile. 
That needs a way tighter manual apply of changes I fear.

File: tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java [hunk 4]
```diff
@@ -45,24 +44,21 @@
  *
  * Prerequisite:
  * The following environment variables must be set prior to running this test:
- * "envconfig_my_int_property" with the value of "45"
- * "ENVCONFIG_MY_BOOLEAN_PROPERTY" with the value of "true"
- * "envconfig_my_string_property" with the value of "haha"
- * "ENVCONFIG_MY_STRING_PROPERTY" with the value of "woohoo"
+ * "my_int_property" with the value of "45"
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java]: wrong direction. Ditching our intermediate fixes that way. so -1

File: tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java [hunk 5]
```diff
@@ -76,17 +72,30 @@ public static Archive deployment() {
     public void checkSetup() {
        //check whether the environment variables were populated by the executor correctly
 
-        if (!"45".equals(System.getenv("envconfig_my_int_property"))) {
-         Assert.fail("Before running this test, the environment variable \"envconfig_my_int_property\" must be set with the value of 45");
+        if (!"45".equals(System.getenv("my_int_property"))) {
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java]: see above. Ditches intermediate fixes. envconfig_.... is correct!

File: tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java [hunk 6]
```diff
@@ -95,30 +104,22 @@ public void checkSetup() {
     public void testPropertyFromEnvironmentVariables() {
         SimpleValuesBean bean = getBeanOfType(SimpleValuesBean.class);
 
-        assertThat(bean.stringProperty, is(equalTo("haha")));
-        assertThat(bean.booleanProperty, is(true));
-        assertThat(bean.intProperty, is(equalTo(45)));
+        // Environment variables are case insensitive on Windows platforms, use Config to
+        // retrieve the "os.name" System property.
+        if(config.getValue("os.name", String.class).contains("Windows")) {
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java]: yikes, that is spooky! We need to find a better TCK way to test this.

File: tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java [hunk 7]
```diff
@@ -95,30 +104,22 @@ public void checkSetup() {
...
     }
 
 
 
     private <T> T getBeanOfType(Class<T> beanClass) {
         return CDI.current().select(beanClass).get();
     }
-
-    @Dependent
-    public static class SimpleValuesBean {
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatchingTest.java]: ditches test we added in ConfigJSR.

File: tck/src/main/java/org/eclipse/configjsr/ImplicitConverterTest.java [hunk 8]
```diff
@@ -19,15 +19,18 @@
...
 package org.eclipse.configjsr;
 
-import javax.config.Config;
+import static org.eclipse.configjsr.base.AbstractTest.addFile;
+
+import java.time.YearMonth;
+
 import javax.inject.Inject;
 
-import org.eclipse.configjsr.converters.implicit.ConvTestTypeWCharSequenceCt;
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/ImplicitConverterTest.java]: also ditches new tests.

File: tck/src/main/java/org/eclipse/configjsr/ImplicitConverterTest.java [hunk 9]
```diff
@@ -36,24 +39,28 @@
...
         JavaArchive testJar = ShrinkWrap
             .create(JavaArchive.class, "implicitConverterTest.jar")
-            .addPackage(ConvTestTypeWCharSequenceCt.class.getPackage())
+            .addPackage(ConvTestTypeWStringCt.class.getPackage())
+            .addClasses(ParseConverterInjection.class, ImplicitConverterTest.class)
             .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
             .as(JavaArchive.class);
 
-        addFile(testJar, "META-INF/javaconfig.properties");
+        addFile(testJar, "META-INF/microprofile-config.properties");
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/ImplicitConverterTest.java]: javaconfig ...

File: tck/src/main/java/org/eclipse/configjsr/ParseConverterInjection.java [hunk 10]
```diff
@@ -16,36 +16,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.eclipse.configjsr.converters.implicit;
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/ParseConverterInjection.java]: I think the git diff messed up this whole file. -1 for merging

File: tck/src/main/java/org/eclipse/configjsr/SimpleValuesBean.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/SimpleValuesBean.java]: Is already an inner class of the test it belongs to. No need to polute thei overall directory imo.

File: tck/src/main/java/org/eclipse/configjsr/converters/implicit/SomeEnumToConvert.java [hunk 12]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2018 Contributors to the Eclipse Foundation
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
```
Comment (struberg) [inline tck/src/main/java/org/eclipse/configjsr/converters/implicit/SomeEnumToConvert.java]: clearly 2018 is more correct

Comment (struberg) [reply tck/src/main/java/org/eclipse/configjsr/converters/implicit/SomeEnumToConvert.java]: But seems the commit messed up the whole file?

File: tck/src/main/resources/internal/META-INF/javaconfig.properties [hunk 13]
```diff
@@ -107,14 +107,14 @@ tck.config.test.javaconfig.converter.urlvalue.broken=tt:--location
 
 
 # implicit Converter tests
-tck.config.test.javaconfig.converter.implicit.enumvalue = FOO
-tck.config.test.javaconfig.converter.implicit.enumvalue.broken = foobar
```
Comment (struberg) [inline tck/src/main/resources/internal/META-INF/javaconfig.properties]: except of different values where is the difference?

**Comment by struberg** [comment]
@Emily-Jiang ping did you see my review? @tomas-langer could you also please review it?

**Comment by tomas-langer** [comment]
I have checked the MR. It seems there are a lot of conflicting changes that should not be merged (as Mark pointed out).
Maybe we should no longer keep microprofile config in sync with config JSR, as there are different drivers/requirements for each and different decisions done?

**Comment by struberg** [comment]
I'll go through the history of changes in mp-config. Maybe it's easier to do this step by step than end2end

**Comment by Emily-Jiang** [comment]
I did step by step approach as suggested. This can be closed.

**Pull request closed without merging**